### PR TITLE
Explicitly set dj name and core network

### PR DESCRIPTION
**Pull request by samredai**
### Summary

This explicitly sets a name (`dj`) and a network (`core`) for the docker compose setups. It allows any other docker container to reliably interact with this docker compose setup by joining the `dj_core` network (docker compose prefixes all networks it creates with the project's name).

### Test Plan

Ran `make check` and `make test`. Also ran `docker compose up` and checked that creating a simple standalone docker container connected to the `dj_core` network could ping the `dj` container.

- [x] PR has an associated issue: #251 
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### Deployment Plan

N/A

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,12 +1,18 @@
-version: "2.2"
-
+version: "3"
+name: "dj"
```
Comment (agorajek) [inline docker-compose.yml:2]: Probably don't need to quote the `dj` value here.

**Review by agorajek — APPROVED** [review]
LGTM, one comment in-line.

File: docker-compose.yml (hunk 1 above)
Comment (samredai) [inline docker-compose.yml:2]:
```suggestion
name: dj
```

Comment (samredai) [reply docker-compose.yml]: Fixed 👍

**Pull request merged**